Tools/AP_Periph/can.cpp

```diff
@@ -2250,6 +2251,12 @@ void AP_Periph_FW::can_rangefinder_update(void)
         // don't send any data
         return;
     }
+    uint32_t sample_ms = rangefinder.last_reading_ms(ROTATION_NONE);
+    if (last_sample_ms == sample_ms) {
```

Comment:
If you're going to do variable rates like this depending on if it's the same sample or not you need to ensure you at least send the duplicate at SOME rate like 1 or more Hz to keep the receiving side healthy/happy.

Reply:
we deliberately don't send dup data - the can node is supposed to not send anything if it gets no data from the rangefinder, making it behave just like a direct attached rangefinder